ci/publishing-images.md

```diff
@@ -20,52 +22,50 @@ The contents of the file should look like this:
+      - master
     tags:
-    # Version tag glob match, start 'v' then a number followed by anything
-    - 'v[0-9]*'
+      - '*'
+    paths:
+      - myapp/version.json
 jobs:
   build:
-    # Only build on redhat-cop repo, do not attempt to build on forks
-    if: github.repository == 'redhat-cop/myrepo'
+    env:
+      context: myapp
+      image_name: myapp
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@master
-    - name: Get image tags
-      id: image_tags
-      run: |
-        echo -n ::set-output name=IMAGE_TAGS::
-        # Tags build from master as latest
-        if [ "${GITHUB_REF}" == 'refs/heads/master' ]; then
-            echo latest
-        # Match git tags to image tags
-        elif [ "${GITHUB_REF:0:10}" == 'refs/tags/' ]; then
-            echo "${GITHUB_REF:10}"
-        # Otherwise, best guess... branch name
-        else
-            echo "${GITHUB_REF/*\//}"
-        fi
-    - name: Build and publish image to Quay
-      uses: elgohr/Publish-Docker-Github-Action@master
-      with:
-        # If quay.io image repository name matches github repository name, then
-        # we can just use the variable, otherwise the name will need to be set
-        # explicitly.
-        name: ${{ github.repository }}
-        username: ${{ secrets.QUAY_USERNAME }}
-        password: ${{ secrets.QUAY_PASSWORD }}
-        registry: quay.io
-        tags: ${{ steps.image_tags.outputs.IMAGE_TAGS }}
-        dockerfile: build/path/Dockerfile
-        context: build/path
+      - uses: actions/checkout@master
+      - name: Get image tags
+        id: image_tags
+        run: |
+          echo -n ::set-output name=IMAGE_TAGS::
+          VERSION=$(jq -r '.version' ${context}/version.json)
+          TAGS=('latest')
+          if [ "${VERSION}" ] && [ "${VERSION}" != "latest" ]; then
+              TAGS+=("${VERSION}")
+          fi
+          if [[ "${GITHUB_REF}" =~ refs/tags/(.*) ]]; then
+              TAGS+=("git-${BASH_REMATCH[1]}")
+          fi
+          ( IFS=$','; echo "${TAGS[*]}" )
+      - name: Build and publish image to Quay
+        uses: docker/build-push-action@v1
+        with:
+          path: ${{ env.context }}
+          registry: ${{ secrets.REGISTRY_SERVER }}
+          repository: ${{ secrets.REGISTRY_NAMESPACE }}/${{ env.image_name }}
+          username: ${{ secrets.REGISTRY_USERNAME }}
+          password: ${{ secrets.REGISTRY_PASSWORD }}
+          tags: "${{ steps.image_tags.outputs.IMAGE_TAGS }}"
 {% endraw %}
 ```
 
+The above also requires a `version.json` file in the directory of your code. The content is very simple, `{"version":"v1.0.0"}` and will be used to tag the image. With this file you can tag your image independently of tags on the repository. This can be convenient when you are building multiple images from the same repo, such as containers-quickstarts.
+
 From there, you can follow our [standard Pull Request](/contrib/) process to get your workflow added to the repo.
```

Comment:
@pabrahamsson I think we should add a note here like:

```
:mag: For security purposes, GitHub does not make secrets accessible to forked repositories. It will therefore not be possible to test workflow code in the context of a Pull Request if the workflow uses secrets. However, you can test the workflow in your own fork by creating the necessary secrets in your forked repo pointing to your own personal registry.
```